tests/helpers/editor-helpers.js

```diff
@@ -29,6 +29,29 @@ export function titleRendered() {
     });
 }
 
+// replaces the title text content with HTML and returns once the HTML has been placed.
+// takes into account converting to plaintext.
+export function replaceTitleHTML(HTML) {
```

Comment:
Rather than re-implenting loops within promises we can rely on Ember's built-in helpers, if we couple those with the `ember-native-dom-helpers` library that we're recommending in integration tests we can simplify this helper quite a bit:

```javascript
import run from 'ember-runloop';
import wait from 'ember-test-helpers/wait';
import {findWithAssert, triggerEvent, focus} from 'ember-native-dom-helpers';

// replaces the title text content with HTML and returns once the HTML has been placed.
// takes into account converting to plaintext.
export function replaceTitleHTML(HTML) {
    let el = findWithAssert('#gh-editor-title div');

    run(() => focus(el));
    run(() => el.innerHTML = HTML);
    run(() => triggerEvent(el, 'input'));
    run(() => triggerEvent(el, 'change'));
    return (window.wait || wait)();
}
```